dali/pipeline/data/tensor_vector.h

```diff
@@ -148,6 +148,26 @@ class TensorVector {
     return type_;
   }
 
+  inline void SetLayout(DALITensorLayout layout) {
+    if (state_ == State::noncontiguous) {
```

Comment:
How about:
```
   if (state_ == State::noncontiguous) {
      DALI_ENFORCE(!tensors_.empty(), "Layout cannot be set uniformly for empty batch");
      for (auto t : tensors_) {
        t->SetLayout(layout);
      }
    } else {
      tl_->SetLayout(layout);
    }   
```

Reply:
I still may need to propagate the layout to individual tensors.